### ISSUE TITLE
Getting the inferred type of a variable?

### ISSUE THREAD

**Issue by Erotemic** [davidhalter/jedi#666]
I'm wondering if it is possible to easily get the type of a variable at the cursor. I'm able to successfully get the completions of a string type and a type inferred through a docstr. Is it possible simply get the inferred variable type?

I have a small test script: 

    import jedi
    import utool as ut
    source = r'''
        def spam(ibs):
            r"""
            Args:
                ibs (ibeis.IBEISController): an object
            """
            import jedi
            jedi.n
            x = ''
            x.l
            ibs.d
        '''
    )
    script = jedi.Script(source, line=9)
    script.completions()
    self = script = jedi.Script(source, line=10)  # NOQA
    script.completions()
    self = script = jedi.Script(source, line=10, column=7)  # NOQA
    script.completions()

(In the above script I use Google style docstrs. I'm using a checkout of jedi that has merged the pull request from https://github.com/WoLpH/jedi to support Google style docstrs. So, completions are working for me in this example.)

I have been able to do some hacking to get what I need by getting the completions and then getting the parent of any functions. 

    cs = script.completions()
    print([c.type for c in cs])
    funcs = [c for c in cs if c.type.startswith('function')]
    f = funcs[0]
    vartype = f.parent().name

This method seems hacky and error prone. I was wondering if there was a better way to do this, or if support for doing this might be added in the future?

**Comment by davidhalter** [comment]
Have you tried `goto_definitions`?

**Comment by Erotemic** [comment]
Ah, that does it. I had tried that but I neglected to put the cursor on the variable of interest. Thank you.

**Comment by davidhalter** [comment]
There's also `jedi.names` if you want to do this for a few names.

**Comment by Erotemic** [comment]
I had tried using jedi.names before. When I did that for the example code I got a list containing one definition: [<Definition def spam>]. I then used the defined_names() method of the first item in that list to get the list

[<Definition ibs>,
 <Definition bar>,
 <Definition import jedi>,
 <Definition x = ''>]

However, I did not see a way to grab the type of an object in that list. I was able to grab the defined names to get the methods of the x variable, however the ibs variable raised "AttributeError: 'Param' object has no attribute 'names_dict'" when I tried to use ibs.defined_names().
